Model/Sync/Reset/Main.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+namespace Yotpo\Core\Model\Sync\Reset;
+
+use Magento\Framework\App\Cache\TypeListInterface;
+use Magento\Framework\App\ResourceConnection;
+use Yotpo\Core\Model\Config;
+
+class Main
+{
+    const DELETE_LIMIT = 10000;
+    const UPDATE_LIMIT = 2000;
+
+    /**
+     * @var ResourceConnection
+     */
+    protected $resourceConnection;
+
+    /**
+     * @var Config
+     */
+    protected $config;
+
+    /**
+     * @var TypeListInterface
+     */
+    protected $cacheTypeList;
+
+    /**
+     * @param ResourceConnection $resourceConnection
+     */
+    public function __construct(
+        ResourceConnection $resourceConnection,
+        Config $config,
+        TypeListInterface $cacheTypeList
+    ) {
+        $this->resourceConnection = $resourceConnection;
+        $this->config = $config;
+        $this->cacheTypeList = $cacheTypeList;
+    }
+
+    /**
+     * @param int $storeId
+     * @param boolean $skipSyncTables
+     * @return void
+     * @throws \Zend_Db_Statement_Exception
+     */
+    public function resetSync($storeId, $skipSyncTables = false)
+    {
+        $this->setResetInProgressConfig($storeId, '1');
+        $this->deleteRunningCronSchedules();
+        if (!$skipSyncTables) {
+            $this->deleteAllFromTables($storeId);
+        }
+    }
+
+    /**
+     * @return array <string>
+     */
+    protected function getTableResourceNames()
+    {
+        return [];
+    }
+
+    /**
+     * @return array <string>
+     */
+    protected function getCronJobCodes()
+    {
+        return [];
+    }
+
+    /**
+     * @return string
+     */
+    protected function getYotpoEntityName()
+    {
+        return '';
+    }
+
+    /**
+     * @param int $storeId
+     * @return void
+     * @throws \Zend_Db_Statement_Exception
+     */
+    private function deleteAllFromTables($storeId)
+    {
+        $tableResourceNames = $this->getTableResourceNames();
+        foreach ($tableResourceNames as $tableResourceName) {
+            $this->deleteAllFromTable($storeId, $tableResourceName);
+        }
+    }
+
+    /**
+     * @param int $storeId
+     * @param string $tableResourceName
+     * @return void
+     * @throws \Zend_Db_Statement_Exception
+     */
+    private function deleteAllFromTable($storeId, $tableResourceName)
```

Comment:
Why not just use `TRUNCATE`?

Reply:
We can't use LIMIT on TRUNCATE. If it is a very huge table and we use TRUNCATE, it will lock that table and there are chances for operational downtime on merchant website. Instead we are using DELETE statements with LIMIT so that we can manage this problem